actions/hadoop-upgrade

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+action-set charmdir.before=${CHARM_DIR}
+export SAVEPATH=$PATH
+. /etc/environment
+export PATH=$PATH:$SAVEPATH
+export JAVA_HOME
+
+current_hadoop_ver=`/usr/lib/hadoop/bin/hadoop version|head -n1|awk '{print $2}'`
+config_hadoop_ver=`config-get hadoop_version`
+new_hadoop_ver=`action-get version`
```

Comment:
A decision was made that the user must configure the units with a version number AND run the action to upgrade. Consider it a safety switch. Discuss @ sync.